### PR TITLE
Fix for inconsistency for JSON type field between mysql and sqlite (#…

### PR DESCRIPTION
**Pull request by KrzysiekNDS**
Issue described in #17713 - there is an inconsistency in json handling between mysql and sqlite

### CODE REVIEW

File: src/Database/Schema/SqliteSchemaDialect.php [hunk 1]
```diff
@@ -141,6 +142,13 @@ protected function _convertColumn(string $column): array
         if (in_array($col, $datetimeTypes)) {
             return ['type' => $col, 'length' => null];
         }
+
+        if (Configure::read('ORM.mapJsonTypeForSqlite') === true) {
+            if (str_contains($col, 'json') && !str_contains($col, 'jsonb')) {
```
Comment (ADmad) [inline src/Database/Schema/SqliteSchemaDialect.php]:
```suggestion
            if (str_contains($col, TableSchemaInterface::TYPE_JSON) && !str_contains($col, 'jsonb')) {
```

**Review by ADmad — APPROVED** [review]

**Comment by othercorey** [comment]
This looks ok to me. I'm not sure about that Configure path. I feel like we should use a Cake prefix for internal configs if we're going to start doing that.

**Comment by ADmad** [comment]
> I feel like we should use a Cake prefix for internal configs if we're going to start doing that.

None of our core configs have a Cake prefix, just the package name.

**Comment by othercorey** [comment]
> > I feel like we should use a Cake prefix for internal configs if we're going to start doing that.
> 
> None of our core configs have a Cake prefix, just the package name.

In that case, this is good. I couldn't remember any ORM examples.

I was thinking of a difference between core app configs that are consumed vis one that's read at runtime like this.

**Comment by ADmad** [comment]
> In that case, this is good. I couldn't remember any ORM examples.

We don't have any other ORM specific config. This one too is temporary to maintain backwards compatibility until we can make the behavior default in 6.0.

**Comment by othercorey** [comment]
Ah, right, as a temporary config this works great.

**Review by othercorey — APPROVED** [review]

**Comment by markstory** [comment]
> We don't have any other ORM specific config. This one too is temporary to maintain backwards compatibility until we can make the behavior default in 6.0.

I think using Configure for backwards compatibility is a great path forward. We can give feature flag style compatibility for upcoming breaking changes. We can also change the default configuration to trigger deprecations or runtime errors.

**Comment by ADmad** [comment]
@KrzysiekNDS Can you please submit a PR to the [5.1 migration guide](https://github.com/cakephp/docs/blob/5.next/en/appendices/5-1-migration-guide.rst) on the `cakephp/docs` repo too adding info about this new feature?

**Pull request merged**